source/ota.c

```diff
@@ -723,7 +727,7 @@ static OtaErr_t setImageStateWithReason( OtaImageState_t stateToSet,
      * If the platform image state couldn't be set correctly, force fail the update by setting the
      * image state to "Rejected" unless it's already in "Aborted".
      */
-    if( ( OTA_PAL_MAIN_ERR( palStatus ) != OtaPalSuccess ) && ( state != OtaImageStateAborted ) )
+    if( ( ( ( palStatus >> OTA_PAL_SUB_BITS ) ) != ( uint32_t ) OtaPalSuccess ) && ( state != OtaImageStateAborted ) )
```

Comment:
why did we change from macro ? And is the type cast of `(uint32_t)` necessary for `OtaPalSuccess` ?